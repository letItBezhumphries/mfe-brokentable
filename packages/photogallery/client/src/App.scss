*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}


html {
  font-size: 62.5%; 
}

body {
  box-sizing: border-box;
}

#Photogallery {
  display: block;
  height: 100vh;
  max-width: 100vw;
}

.main {
  height: 100%;
  max-width: 100%;
  display: block;
}

