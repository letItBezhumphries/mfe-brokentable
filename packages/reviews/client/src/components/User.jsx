import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import "../assets/reviewsStyle.css";

const User = (props) => {
  var vipStatus = props.userData.vip === 'Yes' ? <span className="userVip">{'VIP'}</span> : '';

  var pictureColors = ['#6C8AE4', '#DF4E96', '#D86441', '#bb6acd'];
  var userColor = pictureColors[Math.floor(Math.random() * pictureColors.length)];

  var initials = [];
  for (var i = 0; i < props.userData.username.length; i++) {
    if (props.userData.username[i].match(/[A-Z]/) && initials.length < 2) {
      initials.push(props.userData.username[i]);
    }
  }

  return (
    <div className="user">
      <div className="userPictureContainer">
        <div className="userPicture" style={{backgroundColor: userColor}}>
          <div className="userInitials">{initials.join('').toUpperCase()}</div>
        </div>
      </div>
      <div className="userData">
        <div className="userNameVip">
          {vipStatus}
          <span className="userName">{props.userData.username}</span>
        </div>
        <span className="userLocation">{props.userData.location}</span>
        <div className="userReviewNumberOuter">
          <span className="userReviewNumberInner">
            <FontAwesomeIcon icon={["far", "comment-alt"]} className="commentsCount" />
            {` ${props.userData.reviewNumber} Reviews`}
          </span>
        </div>
      </div>
    </div>
  )
};

export default User;