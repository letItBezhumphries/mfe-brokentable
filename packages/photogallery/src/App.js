import React from 'react';
import { Route } from 'react-router-dom';
import PhotosSection from './components/photosSection/PhotosSection';
import './App.scss';

const App = () => {
  return (
    <div className="main">
      <Route exact path="/restaurants/:id" component={PhotosSection} />
    </div>
  );
};

export default App;
