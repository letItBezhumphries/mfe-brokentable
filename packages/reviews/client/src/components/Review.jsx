import React from 'react';
import User from './User.jsx';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import "../assets/reviewsStyle.css";

const Review = (props) => {
  var stars = [];
  for (var i = 0; i < props.scoreOverall; i++) {
    stars.push(<FontAwesomeIcon key={'score' + i} icon="star" className="filledScore" />);
  }
  for (var i = 0; i < (5 - props.scoreOverall); i++) {
    stars.push(<FontAwesomeIcon key={'emptyScore' + i} icon={["far", "star"]} className="emptyScore" />);
  }

  var months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

  var currentDate = props.date;
  var currentDateArray = props.date.split('-');
  var today = new Date();
  var dinedOn = today.getDate() - Number(currentDateArray[2]);
  
  if (today.getFullYear() - Number(currentDateArray[0]) === 0 && (today.getMonth() + 1) - Number(currentDateArray[1]) === 0 && dinedOn <= 7 && dinedOn > 1) {
    currentDate = `Dined ${dinedOn} days ago`;
  } else if (dinedOn === 1) {
    currentDate = 'Dined a day ago';
  } else if (dinedOn === 0) {
    currentDate = 'Dined today';
  } else {
    currentDate = `Dined on ${months[Number(currentDateArray[1])]} ${Number(currentDateArray[2])}, ${currentDateArray[0]}`
  }

  return (
    <article className="userAndReview">
      <User userData={props.user} />
      <div id="reviewBody" className="reviewBody">
        <div id="reviewScores">
          <div id="reviewStarsDate" className="reviewStarsAndDate">
            <div id="reviewStars">
              {stars}
            </div>
            <div id="reviewDate">
              <span id="date">{currentDate}</span>
            </div>
          </div>
          <div id="otherScores">
            <span id="scoreOverallText" className="reviewText">Overall </span>
            <span id="scoreOverallNumber" className="reviewScore">{props.scoreOverall}</span>
            <span id="scoreFoodText" className="reviewText">Food </span>
            <span id="scoreFoodNumber" className="reviewScore">{props.scoreFood}</span>
            <span id="scoreServiceText" className="reviewText">Service </span>
            <span id="scoreServiceNumber" className="reviewScore">{props.scoreService}</span>
            <span id="scoreAmbienceText" className="reviewText">Ambience </span>
            <span id="scoreAmbienceNumber">{props.scoreAmbience}</span>
          </div>
        </div>
        <div id="reviewText">
          <p>{props.text}</p>
        </div>
      </div>
    </article>
  );
};


export default Review;