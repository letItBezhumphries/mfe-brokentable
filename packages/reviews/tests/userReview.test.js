import React from 'react';
import { shallow } from 'enzyme';
import Review from '../client/src/components/Review.jsx';
import User from '../client/src/components/User.jsx';
import App from '../client/src/components/App.jsx';

describe('It should render individual components', ()=>{
  test('User component renders correctly', () => {
    const wrapper = shallow(<User/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Review component renders correctly', () => {
    const wrapper = shallow(<Review
      date={'2019-02-21'}
      scoreOverall={4}
      scoreFood={4}
      scoreService={3}
      scoreAmbience={5}
      text={'words'}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  test('App component renders correctly', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper).toMatchSnapshot();
  });
})
