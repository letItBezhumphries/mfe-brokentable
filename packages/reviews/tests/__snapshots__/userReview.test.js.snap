// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render individual components App component renders correctly 1`] = `ShallowWrapper {}`;

exports[`It should render individual components Review component renders correctly 1`] = `ShallowWrapper {}`;

exports[`It should render individual components User component renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
